pkg/util/configspec.go

```diff
@@ -188,3 +188,30 @@ func AppendNewExtraConfigValues(
 
 	return append(extraConfig, newExtraConfig...)
 }
+
+// ExtraConfigToMap converts the ExtraConfig to a map with string values.
+func ExtraConfigToMap(input []vimTypes.BaseOptionValue) (output map[string]string) {
+	output = make(map[string]string)
+	for _, opt := range input {
+		if optValue := opt.GetOptionValue(); optValue != nil {
+			// Only set string type values
+			if val, ok := optValue.Value.(string); ok {
+				output[optValue.Key] = val
+			}
+		}
+	}
+	return
+}
+
+// MergeExtraConfig adds the key/value to the ExtraConfig if the key is not present, to let to the value be
```

Comment:
Nit: `to let`

Reply:
amd what do you mean by `The existing usage of ExtraConfig is hard to fit in the reconciliation model.`?

Reply:
These functions and comments are copied from the previous place. Not sure if this statement still holds as it was added a while back ago. Will update the comments to clarify it.